core/shared/src/main/scala/bobcats/Hash.scala

```diff
@@ -17,9 +17,38 @@
 package bobcats
 
 import scodec.bits.ByteVector
+import cats.effect.kernel.Resource
+
+/**
+ * Used for incremental hashing.
+ */
+sealed trait Digest[F[_]] {
```

Comment:
I'm not against this, but I do want to point out some trade-offs ...

An incremental hashing API in terms of `update`/`get`/`reset` is "lower-level" vs an API expressed as a `Stream`.

This is relevant because there is a proposal for browsers to incremental hashing, but this will likely be in the form of streams. I _think_ it might still be possible to implement the API you've declared here with a stream, but we should be thoughtful about that.

https://github.com/wintercg/proposal-webcrypto-streams

Reply:
I agree; I wasn't sure which way we wanted the dependency graph (as in `bobcats <- fs2` or `fs2 -> bobcats` where `->` is dependent on). If the former then more flexibility is always nicer.

Reply:
> `bobcats <- fs2`

Definitely not this. There is a high bar for FS2 taking on new dependencies and although it currently has crypto code I believe moving it out to a library like this makes much more sense.

Reply:
:+1: Then I'll switch over to a `Stream` based API - we can always split into multiple modules if we don't want consumers to pull in `fs2`.